private/buf/cmd/buf/command/mod/modprune/modprune.go

```diff
@@ -102,6 +93,24 @@ func run(
 	}
 	var dependencyModulePins []bufmoduleref.ModulePin
 	if len(requestReferences) > 0 {
+		var remote string
+		if config.ModuleIdentity != nil && config.ModuleIdentity.Remote() != "" {
+			remote = config.ModuleIdentity.Remote()
+		} else {
+			// At this point we know there's at least one dependency. If it's an unnamed module, discover
+			// the right remote from the list of dependencies.
+			remote = bufmoduleref.DiscoverRemote(config.Build.DependencyModuleReferences)
+			container.Logger().Debug(fmt.Sprintf(
+				`%q does not specify a "name". From your dependencies, Buf is defaulting to using remote %q for dependency resolution. This remote may be unable to resolve some of your dependencies. Did you mean to specify a "name: %s/..." on this module?`,
+				existingConfigFilePath,
+				remote, remote,
```

Comment:
Was this meant to be `bufconnect.DefaultRemote, remote`?

Reply:
Not really, we're telling the customer which remote we discovered (not always the default one). Probably the confusion comes from the expression "defaulting to"?

Reply:
Hmm ye. This reads a bit odd because it'll always be the same.. "buf is defaulting to X, and X may not be available. Did you mean to specify X?"